src/sidebar/services/annotations.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const SearchClient = require('../search-client');
+
+// @ngInject
+function annotations(annotationMapper, api, store, streamer, streamFilter) {
+  let searchClient = null;
+
+  /**
+   * Load annotations for all URIs and groupId.
+   *
+   * @param {string[]} uris
+   * @param {string} groupId
+   */
+  function load(uris, groupId) {
+    annotationMapper.unloadAnnotations(store.savedAnnotations());
+
+    // Cancel previously running search client.
+    if (searchClient) {
+      searchClient.cancel();
+    }
+
+    if (uris.length > 0) {
+      searchAndLoad(uris, groupId);
+
+      streamFilter.resetFilter().addClause('/uri', 'one_of', uris);
+      streamer.setConfig('filter', { filter: streamFilter.getFilter() });
+    }
+  }
+
+  function searchAndLoad(uris, groupId) {
+    searchClient = new SearchClient(api.search, {
+      incremental: true,
+    });
+    searchClient.on('results', function(results) {
```

Comment:
For new code prefer arrow functions unless you are creating a method and need to access the value of `this` used to call the function.